description: Out-of-bound access in f2fs_iget() when mounting a corrupted f2fs image
references:
- https://bugzilla.kernel.org/show_bug.cgi?id=200173
- https://git.kernel.org/pub/scm/linux/kernel/git/chao/linux.git/commit/?h=f2fs-dev&id=190286a242344d42aad83c8e70169d6d72a63d05
- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-13098
- https://git.kernel.org/pub/scm/linux/kernel/git/chao/linux.git/commit/?h=f2fs-dev&id=346886775c5fa6a541c0148bbecc0554ab9d6dad
- https://git.kernel.org/pub/scm/linux/kernel/git/jaegeuk/f2fs.git/commit/?h=dev&id=76d56d4ab4f2a9e4f085c7d77172194ddaccf7d2
- https://usn.ubuntu.com/usn/usn-4094-1
comments:
  Debian-bwh: |-
    Introduced in 4.14-rc1 by commit 7a2af766af15 "f2fs: enhance on-disk
    inode structure scalability"
  Ubuntu-sbeattie: |-
    fix commit subject "f2fs: fix to do sanity check with
      extra_attr feature"
  Ubuntu-tyhicks: |-
    The affected user base for this issue is likely small since f2fs is
     not the default filesystem and it is not widely used in Ubuntu
reporters:
- Wen Xu
introduced-by:
  mainline: [7a2af766af15887754f7f7a0869b4603b390876a]
fixed-by:
  linux-4.14.y: [0081c90ebacebb3a82d0d24bf0f42273ce2d902e]
  linux-4.18.y: [b6f493453c78311598fdd204f7815e683818512a]
  mainline: [76d56d4ab4f2a9e4f085c7d77172194ddaccf7d2]
